### ISSUE TITLE
`noUselessIndex` is not applied on project-relative path segments

### ISSUE THREAD

**Issue by park-jemin** [import-js/eslint-plugin-import#2939]
As part of the `no-useless-path-segments`, we have a `noUselessIndex` option to remove unnecessary indexes. So the expected behavior works fine on relative imports:

```ts
import foo from './foo/index'; // flags fine
```

But when it comes to references from project-relative imports (e.g., [absolute imports in Next](https://nextjs.org/docs/app/building-your-application/configuring/absolute-imports-and-module-aliases)), it doesn't seem to get detected:

```ts
import foo from 'foo/index'; // does not flag
```

<img width="548" alt="Screenshot 2023-12-14 at 3 26 52 AM" src="https://github.com/import-js/eslint-plugin-import/assets/59681283/455e6762-8dfd-4331-9410-1a7ceee2db20">

Should we consider making this its own rule?

**Comment by ljharb** [comment]
No, I think this is just part of the tradeoff you make by choosing a nonstandard approach - ie, project-relative imports.

There's no reasonable and reliable way for static analysis to "just know" that bare specifier aliases reliably point to what should be relative file paths.

**Comment by park-jemin** [comment]
Can you elaborate how that would affect detecting unnecessary `index` or `index.js` files? Not sure I'm getting what you mean

**Comment by ljharb** [comment]
A bare specifier typically maps to a package, and packages might have "exports", which wouldn't work with this rule.

**Comment by soryy708** [comment]
Related:

- https://github.com/import-js/eslint-plugin-import/issues/2921
